src/inspectors/package/RustPackageInspector.ts

```diff
@@ -0,0 +1,415 @@
+
+import * as TOML from '@iarna/toml';
+import { DependencyType, Package, PackageVersion } from '../IPackageInspector';
+
+const isRecord = function (v: unknown): v is Record<string, unknown> {
+  return typeof v === 'object' && v !== null;
+};
+const isString = function (v: unknown): v is string {
+  return typeof v === 'string';
+};
+const isOptString = function (v: unknown): v is string | undefined {
+  return typeof v === 'string' || v === undefined;
+};
+const isOptBool = function (v: unknown): v is boolean | undefined {
+  return typeof v === 'boolean' || v === undefined;
+};
+const isOptStringArray = function (v: unknown): v is string[] | undefined {
+  return v === undefined || (Array.isArray(v) && v.every((e) => typeof e === 'string'));
+};
+
+/**
+ * Similar to `lodash.conformsTo` but runs the check on `undefined` instead of failing.
+ */
+const conformsOptional = function (check: Record<string, (v: unknown) => boolean>, value: Record<string, unknown>): boolean {
+  return Object.keys(check).every((key) => check[key](value[key]));
+};
+
+@injectable()
+export class RustPackageInspector extends PackageInspectorBase {
+  private readonly fileInspector: IFileInspector;
+  cargoLock?: CargoLock;
+  cargoManifest?: CargoManifest | CargoWorkspaceManifest;
+
+  constructor(@inject(Types.IFileInspector) fileInspector: IFileInspector) {
+    super();
+
+    this.fileInspector = fileInspector;
+  }
+
+  async init(): Promise<void> {
+    try {
+      this.debug('RustPackageInspector init started');
+
+      const cargoLockString = await this.fileInspector
+        .readFile('Cargo.lock')
+        .catch((err) => (err.code === 'ENOENT' ? Promise.resolve(undefined) : Promise.reject(err)));
+      if (cargoLockString !== undefined) {
+        const cargoLockToml = TOML.parse(cargoLockString);
+        this.cargoLock = RustPackageInspector.parseLock(cargoLockToml);
+      }
+
+      const cargoManifestString = await this.fileInspector
+        .readFile('Cargo.toml')
+        .catch((err) => (err.code === 'ENOENT' ? Promise.resolve(undefined) : Promise.reject(err)));
+      if (cargoManifestString !== undefined) {
+        const cargoManifestToml = TOML.parse(cargoManifestString);
+
+        if ('workspace' in cargoManifestToml) {
+          this.cargoManifest = {
+            workspace: RustPackageInspector.parseWorkspace(cargoManifestToml['workspace']),
+          };
+          this.packages = [];
+        } else {
+          this.cargoManifest = {
+            package: RustPackageInspector.parsePackage(cargoManifestToml['package']),
+            ...RustPackageInspector.parseDependencySet(cargoManifestToml),
+            target: RustPackageInspector.parseTarget(cargoManifestToml['target']),
+            bin: RustPackageInspector.parseBin(cargoManifestToml['bin']),
+            profile: cargoManifestToml['profile'],
+          };
+
+          this.packages = RustPackageInspector.extractPackages(this.cargoManifest, this.cargoLock);
```

Comment:
Keep the convetion and name the method the same as in all the Package inspectors. The code will be much more readable.
```suggestion
          this.packages = RustPackageInspector.addPackages(this.cargoManifest, this.cargoLock);
```